packages/medusa/src/services/batch-job.ts

```diff
@@ -71,6 +136,243 @@ class BatchJobService extends TransactionBaseService<BatchJobService> {
       }
     )
   }
+
+  async validateBatchContext(
+    type: string,
+    context: Record<string, unknown>
+  ): Promise<Record<string, unknown>> {
+    // TODO validate context with batch job strategy corresponding to the given type
+
+    return context
+  }
+
+  async create(data: BatchJobCreateProps): Promise<BatchJob> {
+    return await this.atomicPhase_(async (manager) => {
+      const batchJobRepo: BatchJobRepository = manager.getCustomRepository(
+        this.batchJobRepository_
+      )
+
+      const validatedContext = await this.validateBatchContext(
+        data.type,
+        data.context
+      )
+
+      const toCreate = {
+        status: BatchJobStatus.CREATED,
```

Comment:
todo: remove this; status is inferred so this is not necessary

Reply:
👍